circuits/kimchi/src/polynomials/poseidon.rs

```diff
@@ -65,11 +75,10 @@ pub fn constraint<F: FftField + SquareRootField>() -> E<F> {
 /// ...
 /// The rth position in this array contains the alphas used for the equations that
 /// constrain the values of the (r+1)th state.
+pub fn constraint<F: FftField + SquareRootField>(alphas: &mut impl Iterator<Item = usize>) -> E<F> {
```

Comment:
It's important for the efficiency of the verifier circuit that the poseidon gate has the powers of alpha starting at 0. Can you enforce this somehow?

Reply:
yeah! We just have to declare it as the first constraint in index :) I can add a commit for that

Reply:
Interesting @imeckler - why is that important for efficiency?

Reply:
I believe it's from this snippet in the verifier circuit:

```ocaml
    (* note that for all gates, the powers of alpha start at a^0 = 1 *)
    let poseidon =
      let (pn :: ps) = Vector.rev m.coefficients_comm in
      let res =
        Vector.fold ~init:pn ps ~f:(fun acc c -> c + endoscale acc alpha)
      in
      scale res plonk.poseidon_selector |> negate
```

you have one less endoscale.